ommx_gurobipy_adapter/adapter.py

```diff
@@ -185,38 +191,45 @@ def _set_constraints(self):
             for sos1 in ommx_hints.sos1_constraints:
                 bid = sos1.binary_constraint_id
                 excluded.add(bid)
-                vars = [self.varname_map[str(v)] for v in sos1.decision_variables]
+                vars = [self.varname_map[str(v)] for v in sos1.variables]
                 self.model.addSOS(GRB.SOS_TYPE1, vars)
 
         # Handle regular constraints
-        for constraint in self.instance.raw.constraints:
+        for constraint in self.instance.constraints:
             if constraint.id in excluded:
                 continue
 
-            if constraint.function.HasField("linear"):
-                expr = self._make_linear_expr(constraint.function)
-            elif constraint.function.HasField("quadratic"):
-                expr = self._make_quadratic_expr(constraint.function)
-            elif constraint.function.HasField("constant"):
+            # Check if the constraints is non linear
+            # Non linear are defined as not linear or quadratic.
+            # For more details, refer to https://docs.gurobi.com/projects/optimizer/en/current/reference/python/nlexpr.html
+            for ids, _ in constraint.function.terms.items():
+                if len(ids) >= 3:
+                    raise OMMXGurobipyAdapterError(
+                        f"The constraints must be either `constant`, `linear` or `quadratic`."
+                        f"Constraint ID: {constraint.id}"
+                    )
```

Comment:
```suggestion
            if constraint.function.degree() >= 3:
                raise OMMXGurobipyAdapterError(
                    f"The constraints must be either `constant`, `linear` or `quadratic`."
                    f"Constraint ID: {constraint.id}"
                )
```

こちらも同様にdegree()を使いましょう！